pubky/src/native/internal/pkarr.rs

```diff
@@ -110,6 +110,32 @@ impl Client {
                 _ => None,
             })
     }
+
+    /// Get the homeserver for a given Pubky public key.
+    /// Looks up the pkarr packet for the given public key and returns the content of the first `_pubky` SVCB record.
+    pub async fn get_homeserver(&self, pubky: &PublicKey) -> Option<String> {
+        let packet = self.pkarr.resolve_most_recent(pubky).await?;
+
+        // Check for the `_pubky` SVCB record.
+        let name = format!("_pubky.{}", pubky.to_z32());
+        let maching_names = packet.resource_records(name.as_str()).collect::<Vec<_>>();
+
+        let pubky_records = maching_names
+            .into_iter()
+            .map(|r| r.rdata.clone())
+            .filter(|r| matches!(r, RData::HTTPS(_)))
+            .collect::<Vec<_>>();
+
+        if pubky_records.is_empty() {
+            return None;
+        }
+
+        let record = pubky_records.first().unwrap();
+        if let RData::HTTPS(svc) = record {
+            return Some(svc.target.to_string());
+        }
+        None
```

Comment:
NIT: just a comment here since I already you stance about `match` . I don't even think this is a matter of personal choice or style. This is just not how `rust` is intended to be written. The `match` statement is a cornerstone of safe rust.
```suggestion
        match record {
            RData::HTTPS(svc) => Some(svc.target.to_string()),
            _ => None,
        }
```
In fact, the `is_empty()` and then the `.unwrap()` above are also very far from idiomatic rust
```rust
if pubky_records.is_empty() {
            return None;
        }

        let record = pubky_records.first().unwrap();
```
it all could just be this straight forward to understand and review statement (I only need to look at syntax to know is complete and correct, no even think about un-handled cases).
```rust
pubky_records.first().and_then(|record| match record {
    RData::HTTPS(svc) => Some(svc.target.to_string()),
    _ => None,
})
```
I guess we clash here, but I think I stand with the consensus. Certainly my head has to work 2 times has hard when I read the original code. We can ask other in-house crabs :crab:  to chime in for opinion @tipogi

Reply:
I guess, since `RData` is an enum, using a `match` clause is the way to go for readability and maintainability. However, in this case, where it only handles a specific pattern, I find `if let` to be an acceptable choice here. Still, it does feel a bit odd.

Talking about top snippet (agree with SHA), `.first()` already returns `None` if the vector is empty, so the first check feels unnecessary.

Reply:
The match is literally the same as if let else. There are functionally equivalent and there is zero reason to use the match here.

```rust
match record {
    RData::HTTPS(svc) => Some(svc.target.to_string()),
    _ => None,
}
```

```rust
if let RData::HTTPS(svc) = record {
    return Some(svc.target.to_string());
} else {
   None
}
```

I reused your `Self::extract_host_from_packet` method so there is nothing to do here anymore.

I agree with the `first()?` usage.

Reply:
`if let`, when there is only 2 variants in the Enum and using the `else` is effectively the same as a `match`, indeed.

 I keep supporting it's not OK to avoid `match` in `rust`, specially on `Enums` .